template.yaml

```diff
@@ -321,73 +321,86 @@ Resources:
             Action: sts:AssumeRole
       Path: /
       Policies:
-        - PolicyName: ScopusBucketAccess
+        - PolicyName: NvaScopusBucketAccess
           PolicyDocument:
             Version: '2012-10-17'
             Statement:
               - Effect: Allow
                 Action:
-                  - s3:*
-                Resources: '*'
+                  - s3:GetObject
+                Resource:
+                  - !Sub 'arn:aws:s3:::${ScopusXmlBucket}/*'
 
-  DefaultLambdaPermissions:
-    Type: AWS::IAM::Policy
+  ScopusUnzipperFunctionRole:
+    Type: AWS::IAM::Role
     Properties:
-      PolicyName: !Sub DefaultLambdaPermissions-${AWS::StackName}
-      Roles:
-        - !Ref ReadFromScopusBucketsRole
-      PolicyDocument:
+      AssumeRolePolicyDocument:
         Version: '2012-10-17'
         Statement:
           - Effect: Allow
-            Action:
-              - logs:CreateLogGroup
-              - logs:CreateLogStream
-              - logs:PutLogEvents
-              - lambda:InvokeFunction
-            Resource: "*"
-
+            Principal:
+              Service: lambda.amazonaws.com
+            Action: sts:AssumeRole
+      Path: /
+      Policies:
+        - PolicyName: ScopusUnzipperBucketAccess
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action:
+                  - s3:PutObject
+                Resource:
+                  - !Sub 'arn:aws:s3:::${ScopusZipBucket}/*'
+                  - !Sub 'arn:aws:s3:::${ScopusXmlBucket}/*'
+              - Effect: Allow
+                Action:
+                  - s3:GetObject
+                Resource:
+                  - !Sub 'arn:aws:s3:::${ScopusZipBucket}/*'
```

Comment:
When I test creating stack from template I run into this error:

_Circular dependency between resources: [ScopusUnzipperFunctionUnzipEventPermission, ScopusUnzipperFunction, ScopusZipBucket, ScopusUnzipperFunctionRole]_

However, when not referencing the actual bucket in the policy statements the stack is created: `'arn:aws:s3:::${ScopusZipBucketName}-${AWS::AccountId}/*'`

Reply:
I seem to have used ${} wrong. Will look at this again tomorrow.